src/globus_sdk/services/search/client.py

```diff
@@ -397,6 +397,62 @@ def delete_by_query(
         log.info(f"SearchClient.delete_by_query({index_id}, ...)")
         return self.post(f"/v1/index/{index_id}/delete_by_query", data=data)
 
+    def batch_delete_by_subject(
+        self, index_id: UUIDLike, data: dict[str, t.Any]
+    ) -> response.GlobusHTTPResponse:
+        """
+        Delete data in a Search index as an asynchronous task, deleting multiple
+        documents based on their ``subject`` values.
+
+        A ``task_id`` value will be included in the response.
+
+        :param index_id: The index in which to delete data
+        :type index_id: str or UUID
+        :param data: a batch-delete-by-subject document listing documents to delete
+        :type data: dict
```

Comment:
I think we should accept a list of subjects (list of strings) explicitly here (with additional data params to keep in line with other operations).

No need to force people to look at additional docs to figure out what the post data object needs to look like for something so straightforward.

Reply:
Yeah, I'm okay with that as an interface change.

I'll do that and also expand the test to make an assertion about the payload sent.

Reply:
I think you will like the new tests. It's nothing to `snork` home about, but it gets the job done.

I'm going to mark this as resolved, since the basic thrust of it (allowing `subjects=...` as usage) is covered.